docs/source/notes/serialization.rst

```diff
@@ -2,90 +2,338 @@
-will still share storage after :func:`torch.load`.
+Saving tensors preserves their view relationships:
+
+::
+
+    >>> numbers = torch.arange(1, 10)
+    >>> evens = numbers[1::2]
+    >>> torch.save([numbers, evens], 'tensors.pt')
+    >>> loaded_numbers, loaded_evens = torch.load('tensors.pt')
+    >>> loaded_evens *= 2
+    >>> loaded_numbers
+    tensor([ 1,  4,  3,  8,  5, 12,  7, 16,  9])
+
+Behind the scenes, these tensors share the same "storage." See
+`Tensor Views <https://pytorch.org/docs/master/tensor_view.html>`_ for more
+on views and storage.
+
+When PyTorch saves tensors it saves their storage objects and tensor
+metadata separately. This is an implementation detail that may change in the
+future, but it typically saves space and lets PyTorch easily
+reconstruct the view relationships between the loaded tensors. In the above
+snippet, for example, only a single storage is written to 'tensors.pt'.
+
+In some cases, however, saving the current storage objects may be unnecessary
+and create prohibitively large files. In the following snippet a storage much
+larger than the saved tensor is written to a file:
 
 ::
 
-    >>> tensor = torch.zeros(1000000)
-    >>> slice1 = tensor[:1000]
-    >>> slice2 = tensor[:10] # slice1 and slice2 share the same storage
-    >>> torch.save([slice1, slice2], 'share.pt')
-    >>> loaded_1, loaded_2 = torch.load('share.pt')
-    >>> loaded_1[0]
-    tensor(0.)
-    >>> loaded_2[0]
-    tensor(0.)
-    >>> loaded_2[0] = 1
-    >>> loaded_1[0] # loaded tensors still share storage
-    tensor(1.)
-
-Note that saving storage instead of tensor itself means the serialized file size might not match tensor size.
-In the example above the whole `tensor`'s storage (of size 1000000) is serialized instead of only slices.
-When tensor is expanded from a smaller storage, serialized file size might be smaller than tensor size as well.
+    >>> large = torch.arange(1, 1000)
+    >>> small = large[0:5]
+    >>> torch.save(small, 'small.pt')
+    >>> loaded_small = torch.load('small.pt')
+    >>> loaded_small.storage().size()
+    999
+
+Instead of saving only the five values in the `small` tensor to 'small.pt,'
+the 999 values in the storage it shares with `large` were saved and loaded.
+
+When saving tensors with fewer elements than their storage objects, the size of
+the saved file can be reduced by first cloning the tensors. Cloning a tensor
+produces a new tensor with a new storage object containing only the values
+in the tensor:
+
+::
+
+    >>> large = torch.arange(1, 1000)
+    >>> small = large[0:5]
+    >>> torch.save(small.clone(), 'small.pt')  # saves a clone of small
+    >>> loaded_small = torch.load('small.pt')
+    >>> loaded_small.storage().size()
+    5
+
+Since the cloned tensors are independent of each other, however, they have
+none of the view relationships the original tensors did. If both file size and
+view relationships are important when saving tensors smaller than their
```

Comment:
nit: "construct new storage objects" sounds like something the user should be doing.  Better to say something like "construct new tensors with the desired view relationships before saving, such that the storage object sizes are minimized."

Reply:
Sure. Sounds good.